### DISCUSSION TITLE
How do logs work?

### DISCUSSION THREAD

**Discussion by cknight** [Hexagon/pup#48] [Q&A] [answered]
Do the logs continually grow without limit?  I have an application which logs lots of data and I want to ensure I can leave pup running for a significant time without any manual intervention required to clean up logs.

**Comment by Hexagon** [comment, marked as answer]
The internal logs (stored for use by command `pup logs` and possibly plugins) is configurable between 0 and 366 hours (about two weeks), see [this](https://pup.56k.guru/usage/configuration/#logger) section. 

If you configure pup to log to a file, it will grow forever as pup has no internal mechanism for log rotation.

**Comment by vuau** [comment]
where does the `pup logs` store log files if I haven't configure paths for them?
